external/kafka/src/main/scala/org/apache/spark/streaming/kafka/DirectKafkaInputDStream.scala

```diff
@@ -118,6 +123,7 @@ class DirectKafkaInputDStream[
       context.sparkContext, kafkaParams, currentOffsets, untilOffsets, messageHandler)
 
     currentOffsets = untilOffsets.map(kv => kv._1 -> kv._2.offset)
+    offsetMap += ((validTime, currentOffsets))
```

Comment:
I don't think offsetMap is required at all. The offset ranges are stored in the KafkaRDDs. Why not just pull up the offset from them, rather than introducing another hashmap that needs to be managed/cleared?